js/views/conversation_view.js

```diff
@@ -201,11 +201,45 @@
             this.$('.bottom-bar form').addClass('active');
         },
 
+        updateUnread: function() {
+            this.updateLastSeenIndicator();
+            this.model.markRead();
+        },
+
         onOpened: function() {
             this.view.resetScrollPosition();
             this.$el.trigger('force-resize');
             this.focusMessageField();
-            this.model.markRead();
+
+            if (this.inProgressFetch) {
+                this.inProgressFetch.then(this.updateUnread.bind(this));
+            } else {
+                this.updateUnread();
+            }
```

Comment:
That would be cleaner, but I think it would leave us open to race conditions, since a single `.then()` here would always resolve async. Between the time this code runs and the `.then()` calls our provided function, the promise could be updated.